src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -321,6 +321,99 @@ def validate_select(namespace):
 
 # region COMMAND VALIDATORS
 
+
+def process_download_batch_parameters(namespace):
+    """Process the parameters for storage blob download command"""
+    from azure.cli.command_modules.storage.storage_url_helpers import parse_url
+
+    # 1. quick check
+    if namespace.destination is None:
+        raise ValueError('Destination parameter is missing.')
+
+    if namespace.source is None:
+        raise ValueError('Source parameter is missing.')
+
+    if not os.path.exists(namespace.destination) or not os.path.isdir(namespace.destination):
+        raise ValueError('Destination folder {} does not exist'.format(namespace.source))
+
+    # 2. try to extract account name and container name from source string
+    storage_desc = parse_url(namespace.source)
+
+    if storage_desc.blob is not None:
+        raise ValueError('incorrect usage: --source should be either container URL or name')
+
+    if namespace.account_name is None:
+        namespace.account_name = storage_desc.account
+
+    namespace.source_container_name = storage_desc.container
+
+
+def process_upload_batch_parameters(namespace):
+    """Process the source and destination of storage blob upload command"""
+    from azure.cli.command_modules.storage.storage_url_helpers import parse_url
+
+    # 1. quick check
+    if namespace.destination is None:
+        raise ValueError('incorrect usage: --destination is missing.')
+
+    if namespace.source is None:
+        raise ValueError('incorrect usage: --source is missing.')
+
+    if not os.path.exists(namespace.source):
+        raise ValueError('incorrect usage: source {} does not exist'.format(namespace.source))
+
+    if not os.path.isdir(namespace.source):
+        raise ValueError('incorrect usage: source must be a directory')
+
+    # 2. try to extract account name and container name from destination string
+    storage_desc = parse_url(namespace.destination)
+
+    if storage_desc.blob is not None:
+        raise ValueError('incorrect usage: destination cannot be a blob url')
+
+    namespace.destination_container_name = storage_desc.container
+
+    if not namespace.account_name:
+        namespace.account_name = storage_desc.account
+
+    # 3. collect the files to be uploaded
+    namespace.source = os.path.realpath(namespace.source)
+
+    if namespace.pattern:
+        from fnmatch import fnmatch
+        pattern = os.path.join(namespace.source, namespace.pattern.lstrip('/'))
+
+        def _file_selector(file_path):
+            return fnmatch(file_path, pattern)
+    else:
+        def _file_selector(_):
+            return True
```

Comment:
this is when pattern is not given, therefore all files are matched. this will avoid spend time on matching.